dev/devicelab/lib/tasks/perf_tests.dart

```diff
@@ -1236,6 +1287,11 @@ class PerfTest {
         _addOpenGLESToManifest(testDirectory);
         manifestReset = () => _resetManifest(testDirectory);
       }
+      Future<void> Function()? plistReset;
+      if (disablePartialRepaint) {
+        _disablePartialRepaint(testDirectory);
```

Comment:
If this throws, `_resetManifest()` won't be run.

Reply:
Maybe it won't matter in this case because you can't have `disablePartialRepaint` and `forceOpenGLES` both true at the same time, but it'd be nice to find a better pattern to follow here for when the third thing comes along...

Reply:
I think the reset commands should always be safe to run. Let me re-arrange this a bit.

Reply:
okay re-arrangement concluded. We should always reset if we throw now.